jax/_src/lib/__init__.py

```diff
@@ -128,13 +128,31 @@ def _xla_gc_callback(*args):
 # TODO(rocm): check if we need the same for rocm.
 
 def _cuda_path() -> str | None:
-  _jaxlib_path = pathlib.Path(jaxlib.__file__).parent
   # If the pip package nvidia-cuda-nvcc-cu11 is installed, it should have
   # both of the things XLA looks for in the cuda path, namely bin/ptxas and
   # nvvm/libdevice/libdevice.10.bc
-  path = _jaxlib_path.parent / "nvidia" / "cuda_nvcc"
-  if path.is_dir():
-    return str(path)
+
+  def _try_jaxlib_relative_path() -> str | None:
+    """Try to find the path to the cuda_nvcc directory relative to jaxlib."""
+    path = pathlib.Path(jaxlib.__file__).parent / "nvidia" / "cuda_nvcc"
+    if path.is_dir():
+      return str(path)
+    return None
+
+  def _try_direct_cuda_nvcc_path() -> str | None:
+    """Try to import `cuda_nvcc` and getting its path directly."""
+    try:
+      from nvidia import cuda_nvcc
+    except ImportError:
+      return None
+    cuda_nvcc_path = pathlib.Path(cuda_nvcc.__file__).parent
+    return str(cuda_nvcc_path)
+
+  if (path := _try_jaxlib_relative_path()) is not None:
```

Comment:
I think I'd be comfortable simply removing the jaxlib-relative path. It never was the right thing to do, given, for example, you might have installed `cuda_nvcc` in the system package root but `jaxlib` in a different package root.

Reply:
Sounds good! I'll remove it. Do you think it would make sense to add a check for `$CUDA_ROOT`? That should fix Filippo's issue in #22590. I'm happy to add that here while we're at it.

Reply:
I just pushed the `$CUDA_ROOT` change. Let me know if anything needs to be tweaked.